src/IIIFPresentation/API/Converters/ManifestConverter.cs

```diff
@@ -42,12 +46,71 @@ public static PresentationManifest SetGeneratedFields(this PresentationManifest
         iiifManifest.Slug = hierarchy.Slug;
         iiifManifest.PaintedResources = dbManifest.GetPaintedResources(pathGenerator, assets);
         iiifManifest.Space = pathGenerator.GenerateSpaceUri(dbManifest)?.ToString();
+
+        // Note ??= - this is only if we don't yet have Items set by background process
+        iiifManifest.Items ??= GenerateProvisionalItems(iiifManifest.PaintedResources);
+        
         iiifManifest.EnsurePresentation3Context();
         iiifManifest.EnsureContext(PresentationJsonLdContext.Context);
         
         return iiifManifest;
     }
 
+    private static List<Canvas>? GenerateProvisionalItems(List<PaintedResource>? paintedResources)
+    {
+        if (paintedResources is not {Count: > 0})
+            return null;
+
+        return paintedResources
+            .GroupBy(pr => pr.CanvasPainting.CanvasOrder)
+            .Select(GenerateProvisionalCanvas)
+            .ToList();
+
+        Canvas GenerateProvisionalCanvas(IGrouping<int?, PaintedResource> canvasPaintings)
+        {
+            var canvasId = canvasPaintings.First().CanvasPainting.CanvasId;
+            var c = new Canvas
+            {
+                Id = canvasId,
+                Items =
+                [
+                    new()
+                    {
+                        Id = $"{canvasId}/page",
```

Comment:
These Ids will persist to 'final' manifest in #231 so be good to get the formats in ticket output now as it should make persisting them easier.